pkg/deploy/controller/deployment/controller.go

```diff
@@ -144,13 +144,18 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Generate a deployer pod spec.
 		deployerPod, err := c.makeDeployerPod(deployment)
 		if err != nil {
-			return fatalError(fmt.Sprintf("couldn't make deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
+			nextStatus = deployapi.DeploymentStatusFailed
+			updatedAnnotations[deployapi.DeploymentStatusReasonAnnotation] = err.Error()
+			c.emitDeploymentEvent(deployment, kapi.EventTypeWarning, "FailedDecoding", err.Error())
+			break
 		}
 		// Create the deployer pod.
 		deploymentPod, err := c.pn.Pods(deployment.Namespace).Create(deployerPod)
-		// Retry on error.
 		if err != nil {
-			return actionableError(fmt.Sprintf("couldn't create deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
```

Comment:
Not sure what happened to my previous answer...

> A create error could be transient if using GenerateName.

We don't use GenerateName for deployer pods so that's fine.

> quota temporarily exceeded

We were just discussing this in Trello w/ @danmcp in https://trello.com/c/YjKyBsju/669-0-deployments-shouldn-t-consume-ongoing-quota. Long story short, we want to gracefully handle temporarily exceeded quota by not retrying to create and fail.